python/pyiceberg/catalog/hive.py

```diff
@@ -0,0 +1,262 @@
+    DoubleType: "double",
+    DateType: "date",
+    TimeType: "string",
+    TimestampType: "timestamp",
+    StringType: "string",
+    UUIDType: "string",
+    BinaryType: "binary",
+    FixedType: "binary",
+    DecimalType: None,
+    StructType: None,
+    ListType: None,
+    MapType: None,
+}
+
+
+class _HiveClient:
+    _transport: TTransport
+    _client: Client
+
+    def __init__(self, url: str):
+        url_parts = urlparse(url)
+        transport = TSocket.TSocket(url_parts.hostname, url_parts.port)
+        self._transport = TTransport.TBufferedTransport(transport)
+        protocol = TBinaryProtocol.TBinaryProtocol(transport)
+
+        self._client = Client(protocol)
+
+    def __enter__(self) -> Client:
+        self._transport.open()
+        return self._client
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._transport.close()
+
+
+class HiveCatalog(Catalog):
+    _client: _HiveClient
+
+    def __init__(self, name: str, properties: Properties, url: str):
+        super().__init__(name, properties)
+        self._client = _HiveClient(url)
+
+    def _storage_descriptor(self, schema: Schema, location: Optional[str]) -> StorageDescriptor:
+        ser_de_info = SerDeInfo(serializationLib="org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe")
+        return StorageDescriptor(
+            self._columns(schema),
+            location,
+            "org.apache.hadoop.mapred.FileInputFormat",
+            "org.apache.hadoop.mapred.FileOutputFormat",
+            serdeInfo=ser_de_info,
+        )
+
+    def _columns(self, schema: Schema) -> List[FieldSchema]:
+        return [FieldSchema(field.name, self._convert_hive_type(field.field_type), field.doc) for field in schema.fields]
+
+    def _convert_hive_type(self, col_type: IcebergType) -> str:
+        if hive_type := hive_types.get(type(col_type)):
+            return hive_type
+        raise NotImplementedError(f"Not yet implemented column type {col_type}")
+
+    def _convert_hive_into_iceberg(self, _: HiveTable) -> Table:
+        # Requires reading the manifest, will implement this in another PR
+        # Also nice to have the REST catalog in first
+        return Table()
+
+    def create_table(
+        self,
+        identifier: Union[str, Identifier],
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: Optional[PartitionSpec] = None,
+        properties: Optional[Properties] = None,
+    ) -> Table:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        current_time_millis = int(time())
+        tbl = HiveTable(
+            dbName=database_name,
+            tableName=table_name,
+            owner=getuser(),
+            createTime=current_time_millis // 1000,
+            lastAccessTime=current_time_millis // 1000,
+            sd=self._storage_descriptor(schema, location),
+            tableType="EXTERNAL_TABLE",
+            parameters={"EXTERNAL": "TRUE"},
```

Comment:
This is missing the parameters from [`set_parameters`](https://github.com/apache/iceberg/blob/master/python_legacy/iceberg/hive/hive_table_operations.py#L132-L137) in the legacy client. Those two are important because they identify the table as Iceberg and set the metadata location.

Looks like this still needs to use `FileIO` and create a file in the metadata location.

Reply:
Yes, the write path is not fully implemented (yet). Also, the Hive locking is not in there yet. I've updated the code to set the right properties on the table. I'll implement creating the JSON file in a separate PR.

Reply:
Agreed on the metadata JSON file. Sounds like a good plan.